src/parser/mage/arcane/modules/features/RuleOfThrees.js

```diff
@@ -24,7 +24,7 @@ class RuleOfThrees extends Analyzer {
 		if (spellId !== SPELLS.ARCANE_BARRAGE.id) {
 			return;
 		}
-		if (this.selectedCombatant.hasBuff(SPELLS.RULE_OF_THREES_BUFF.id)) {
+		if (this.selectedCombatant.hasBuff(SPELLS.RULE_OF_THREES_BUFF.id,event.timestamp + 1)) {
```

Comment:
This works for now, but just as a reminder we can fix this with a normalizer as well I think

Reply:
the order of the events isnt the problem. If it was, I would have just made a normalizer to rearrange them.

As you can see here, the Rule of Threes buff was removed BEFORE the Arcane Barrage cast, and yet it still said he had Rule of Threes when Arcane Barrage was cast
https://media.discordapp.net/attachments/499995593465135104/500054178962604053/unknown.png

Reply:
Wouldnt the normalizer work if you didnt use the timestamp?